pulpcore/app/serializers/fields.py

```diff
@@ -193,8 +193,12 @@ class BaseURLField(serializers.CharField):
     """
 
     def to_representation(self, value):
-        if settings.CONTENT_HOST:
-            host = settings.CONTENT_HOST
-        else:
-            host = self.context['request'].get_host()
-        return ''.join([host, settings.CONTENT_PATH_PREFIX, value])
+        base_path = value
+        host = settings.CONTENT_HOST or ''
```

Comment:
This one defaults to `None` in settings.py. That default was chosen to align with this code. What about having this PR remove the `or ''` and instead setting `CONTENT_HOST = ''` in settings.py?

Reply:
There isn't anything preventing a user to setting it to `None` which we should deal with gracefully.

Reply:
I don't think handling this gracefully is beneficial to the user. In fact I think it masks situations which should be explicit for the user. Here are some of the reasons I think about.

If the user is applying an incorrect config we should not translate it into another value and we doubly shouldn't do so behind the scenes. To translate for them allows their errors to pass unnoticed and silently. Say they used a None value instead of a string, they should be presented this mistake immediately so they can fix it.

Also if we're concerned about users setting invalid values we should have some sort of validation for our settings. If we do that, we would also not want it here because it wouldn't be centralized. Doing it everywhere it is accessed creates a DRY problem which spreads out the default value.

Reply:
I think `None` and `''` are equivalent in this case.  Forcing the user to use `''` instead of `None` does not seem valuable.  I think we should focus on the user's intent and not bog them down in syntax.

Reply:
+1 for validating the settings but raising an error on `None` instead of `''` seems pedantic.

Reply:
So the question is whether for `CONTENT_HOST`, the value `None` is equivalent to `''` or whether it should be considered erroneous?

Reply:
My request is that we don't do interpretation or adjustment of these variables here. I hope this code can only deal with how these config values are brought together without adjusting their possible values along the way.

Reply:
> My request is that we don't do interpretation or adjustment of these variables here.

That seems reasonable to me.

Reply:
For the purpose of generating the URL here, I don't see how `None` and `''` could be anything but equivalent.  So, not interpreting or adjusting but rather just adds robustness.

Reply:
I don't disagree. I just wonder if there could be a more central place to do it?

Reply:
Perhaps.  Where?
Do we really want to develop and maintain a settings validation/conditioning mechanism  to just handle `settings.CONTENT_HOST or ''`?